rust/otap-dataflow/crates/otap/src/experimental/gigla_exporter/config.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashMap;
+
+/// Configuration for the GigLA Exporter matching the Collector's schema.
+#[derive(Debug, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    /// API configuration for GigLA
+    pub api: ApiConfig,
+
+    /// Authentication configuration
+    #[serde(default)]
+    pub auth: AuthConfig,
+}
+
+/// Authentication method for Azure
+#[derive(Debug, Deserialize, Clone, PartialEq, Default)]
+#[serde(rename_all = "lowercase")]
+pub enum AuthMethod {
+    /// Use Managed Identity (system or user-assigned with client_id)
+    #[serde(alias = "msi", alias = "managed_identity")]
+    #[default]
+    ManagedIdentity,
```

Comment:
Are we planning to support `Azure Arc` authentication for Edge devices? Azure Arc-enabled AKS requires different [authentication logic](https://learn.microsoft.com/en-us/azure/azure-arc/servers/managed-identity-authentication). If it's on the roadmap, we can add a TODO for now, else ignore this.

Reply:
I'd like to remove any parts of this PR that are dead code. If ManagedIdentity is not used, why introduce it now?